File: +migration/read-signup.py
import mysql.connector

def read_data():
    data_list = []
    try:
        # Establishing the connection to MySQL
        connection = mysql.connector.connect(
            host="localhost",
            user="root",
            password="",
            database="gripdespro"
        )

        # Creating a cursor object using the cursor() method
        cursor = connection.cursor()

        # SQL query to select all data from the "grip_signup" table
        select_query = "SELECT * FROM grip_signup"

        # Execute the SQL query
        cursor.execute(select_query)

        # Fetch all the rows
        rows = cursor.fetchall()

        # Append each row to the data list
        for row in rows:
            # Convert each row to a list and append to data_list
            data_list.append([str(row[0])] + list(row)[1:])

    except mysql.connector.Error as error:
        print(f"Error: {error}")

    finally:
        # Closing the connection
        if connection.is_connected():
            cursor.close()
            connection.close()
    
    return data_list

# Example usage
data = read_data()
print(data)

File: api/read-signup.php
<?php

$host = "localhost";
$user = "root";
$password = "";
$database = "gripdespro";

try {
    // Establishing the connection to MySQL
    $connection = mysqli_connect($host, $user, $password, $database);

    if (!$connection) {
        die("Connection failed: " . mysqli_connect_error());
    }

    // SQL query to select all data from the "grip_signup" table
    $select_query = "SELECT * FROM grip_signup";

    // Execute the SQL query
    $result = mysqli_query($connection, $select_query);

    if (mysqli_num_rows($result) > 0) {
        echo "<table border='1'>";
        echo "<tr><th>ID</th><th>Firstname</th><th>Lastname</th><th>Password</th></tr>";
        // Output data of each row
        while ($row = mysqli_fetch_assoc($result)) {
            echo "<tr>";
            echo "<td>" . $row["id"] . "</td>";
            echo "<td>" . $row["firstname"] . "</td>";
            echo "<td>" . $row["lastname"] . "</td>";
            echo "<td>" . $row["password"] . "</td>";
            echo "</tr>";
        }
        echo "</table>";
    } else {
        echo "No data found";
    }

    // Free result set
    mysqli_free_result($result);
    // Closing the connection
    mysqli_close($connection);
} catch (Exception $e) {
    echo "Error: " . $e->getMessage();
}

?>
